File: packages/backend/functions/signin/types/messages.go
package types

type BaseMessage struct {
	Type   string `json:"type"`
	ItemId string `json:"itemId"`
}

type ErrorMessage struct {
	BaseMessage
	Message     string `json:"message"`
	ReloadUsers bool   `json:"reloadUsers"`
}

type RenderProgressMessage struct {
	BaseMessage
	Percent float32 `json:"percent"`
}

type RenderSuccessMessage struct {
	BaseMessage
	VideoUrl string  `json:"videoUrl"`
	Elapsed  float64 `json:"elapsed"`
}

File: packages/frontend/src/types/messages.ts
import { ErrorBody } from "./api";

export type BaseMessage = {
  type: "error" | "progress" | "success";
  itemId: string;
  timestamp: number;
};

export type ErrorMessage = BaseMessage & {
  type: "error";
  message: ErrorBody;
  reloadUsers: boolean;
};

export type RenderProgressMessage = BaseMessage & {
  type: "progress";
  percent: number;
};

export type RenderSuccessMessage = BaseMessage & {
  type: "success";
  videoUrl: string;
  elapsed: number;
};
